rllib/policy/tf_policy.py

```diff
@@ -330,18 +342,35 @@ def _initialize_loss(self, loss: TensorType,
             self._loss_input_dict["state_in_{}".format(i)] = ph
 
         if self.model and not isinstance(self.model, tf.keras.Model):
-            self._loss = self.model.custom_loss(loss, self._loss_input_dict)
+            self._losses = self.model.custom_loss(losses,
+                                                  self._loss_input_dict)
             self._stats_fetches.update({"model": self.model.metrics()})
         else:
-            self._loss = loss
-
-        if self._optimizer is None:
-            self._optimizer = self.optimizer()
-        self._grads_and_vars = [
-            (g, v) for (g, v) in self.gradients(self._optimizer, self._loss)
-            if g is not None
-        ]
-        self._grads = [g for (g, v) in self._grads_and_vars]
+            self._losses = losses
+        # Backward compatibility.
+        self._loss = self._losses[0] if self._losses is not None else None
+
+        if not self._optimizers:
+            self._optimizers = force_list(self.optimizer())
+            # Backward compatibility.
+            self._optimizer = self._optimizers[0] if self._optimizers else None
+
+        # Supporting more than one loss/optimizer.
+        if self.config["_tf_policy_handles_more_than_one_loss"]:
+            self._grads_and_vars = []
+            self._grads = []
+            for group in self.gradients(self._optimizers, self._losses):
+                g_and_v = [(g, v) for (g, v) in group if g is not None]
+                self._grads_and_vars.append(g_and_v)
+                self._grads.append([g for (g, v) in g_and_v])
```

Comment:
don't know if our linter will catch unused variable. since v is not used here:

[g for (g, _) in g_nd_v]

same below.

Reply:
+1

Reply:
done